5.OAM_KPT_Demo/README.md

```diff
@@ -0,0 +1,216 @@
+
+Both kpt and OAM are outcomes of "Configuration-as-Data (or, Infrastructure-as-Data, IaD)".
+
+Pioneered by [Kubernetes community](https://twitter.com/bgrant0607/status/1221485437153243137), Configuration-as-Data emphasizes that "configuration should be treated as data and leverage pipelines for manipulation and policy enforcement". In Kubernetes, Configuration-as-Data approach builds upon the design of the [Kubernetes resource model (KRM)](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/architecture/resource-management.md). As a result, today any resource we applied to Kubernetes is a piece of data that represents the desired state for certain part of application or infrastructure in the real system.
+
+With the heart of Configuration-as-Data, a typical Kubernetes native application management workflow just looks like a "pipeline". See the picture below:
+
+![img](./resource/kpt_oam.png)
+
+In this workflow, kpt is the manipulator of data. Stored in data source like Git, the original data (e.g. deployment.yaml) will pass through a pipeline of kpt functionalities to be manipulated into the desire state step by step. For example, `labels` added, `replicas` modified and `image` updated etc. 
+
+So what is OAM then? OAM is the format of data. More accurately, OAM is a specific data format so Kubernetes can expose higher level abstraction such as "application" to developers.
+
+In that sense, it's a natural match for using kpt to manipulate the data which is formatted by OAM specification.
+
+## Pre-requisites
+
+Install OAM Kubernetes runtime (Crossplane) by following [its installation guides](https://github.com/oam-dev/crossplane-oam-sample#installation).
+
+## Create App Repository for kpt
+
+kpt directly use GitHub repo as App Repository, so no action needed.
+
+With the help of kpt, we could directly use GitHub Repo as App Repository without organizing apps in any fixed format.
+
+### Release your application to App Repository
+
+So release your OAM app only needs two steps.
+
+1. Create/Fork a github repo.
+
+2. Commit and push your app.
+    ```shell
+    git add sampleapp/
+    git commit -m "my sampleapp commit"
+    git remote add origin git@github.com:<your-account>/<your-app-repo>.git
+    git push -u origin master
+    ```
+
+### Fetch OAM app from remote Repository
+
+Using our [example repository](https://github.com/oam-dev/sample/tree/master/5.OAM_KPT_Demo/repository/) for this demo.
+
+You could fetch OAM app from remote Repository using [kpt pkg get](https://googlecontainertools.github.io/kpt/reference/pkg/get/).
+
+#### Command
+
+```shell
+kpt pkg get https://github.com/oam-dev/samples.git/5.OAM_KPT_Demo/repository/sampleapp sampleapp
+```
+
+#### Output
+
+```output
+fetching package /5.OAM_KPT_Demo/repository/sampleapp from https://github.com/oam-dev/samples to sampleapp
+```
+
+```shell
+➜  kpt tree sampleapp
+sampleapp
+├── Kptfile
+├── appconfig.yaml
+├── component.yaml
+├── traitdef.yaml
```

Comment:
XxxDefinition should not be part of the app. Please fix this and same issues below.

Reply:
sorry, fixed